### PR TITLE
Exclude patterns

### PR DESCRIPTION
**Pull request by sasha-stratoscale**
Added exclude patterns to exclude specific packages and join with site packages at runtime

### CODE REVIEW

**Comment by motty-stratoscale** [comment]
Does it mean that I can have a package named strato.something in site-packages, and any project that wants to use it (in addition to its "private" packages that also begin with "strato") will have a file like:
/usr/share/strato-api/strato/__init__.py   that does the join trick?

@shimshon-stratoscale  If it's true, we don't need to give strato_kv a special namespace,

**Comment by sasha-stratoscale** [comment]
yep .. all you need is to specify the --joinWithSitePackages=strato during the build stage when  you  packing the eggs

**Comment by motty-stratoscale** [comment]
And how the tools on developer machines (pylint, whiteboxtest) will see these packages? By upseto deps?
E.g. if snld uses pycommonbla, which is going to be a site package, then:
1. snld will have upseto dep on pycommonbla.
2. snld packaging will do   
upseto.packegg --joinWithSitePackages=strato  --excludeModule=strato.common.bla
?

**Comment by sasha-stratoscale** [comment]
why to you want to pylint stuff that is already installed under site package ?

**Comment by motty-stratoscale** [comment]
I don't pylint the code in site packages (being there is the ultimate Kosher stamp), but the code that uses it.

**Comment by sasha-stratoscale** [comment]
it wont find it .. the "path" extention is only happening inside the egg .. and if i remember correctly you are pylinting source files

**Comment by motty-stratoscale** [comment]
Maybe you're right and pylint ignores everything global. I just ran it on  "import math; x = math.sin(1, 2)" and it was totally silent.  (Running the code said:   TypeError: sin() takes exactly one argument (2 given)

BUT I want to use pycommonbla to 1001 other things on my machine or CI machine.
For example in a racktest:
result = host1.ssh.run.script("...")
TS_ASSERT_EQUALS(result, pycommonbla.consts.pi)

How to import pycommonbla.consts.pi?
1. Monkey upseto-depends on pycommonbla?
2. Every developer & CI machine will have pycommonbla in its site packages?
3. solvent bring pycommonbla to some directory, and add it to PYTHONPATH? (That's what I planned for strato_kv)
Thanks

**Comment by sasha-stratoscale** [comment]
Short answer is that it is up to the developer how he uses the feature  :)
There is an advantage having stuff installed in site package, (you dont
need to pack them along with your other stuff) [this is the way that we use
the upseto itself :) and libvirt and everything else]
There is an abvious disadvantage in doing it during the development stage
for the code that you may change, which is  that in order to test the code
you need to go through the long cycle of making 1 line change/ make/commit
(change the hash)/deploy/fetch/install/test ..  (of course you can automate
the stuff, however it will always imply that you will rebuild the
dependency from scratch and install it .. ), while packing it with your own
code will result it : do 1 line of code -> test it :)
IMHO first one is for final product assembly so that there will be the same
version of package across all stratoscale (if needed), while the later is
best suite during development/test for faster response cycles ..








On Thu, Jun 25, 2015 at 6:42 PM, motty-stratoscale <notifications@github.com
> wrote:

> Maybe you're right and pylint ignores everything global. I just ran it on
> "import math; x = math.sin(1, 2)" and it was totally silent. (Running the
> code said: TypeError: sin() takes exactly one argument (2 given)
>
> BUT I want to use pycommonbla to 1001 other things on my machine or CI
> machine.
> For example in a racktest:
> result = host1.ssh.run.script("...")
> TS_ASSERT_EQUALS(result, pycommonbla.consts.pi)
>
> How to import pycommonbla.consts.pi?
> 1. Monkey upseto-depends on pycommonbla?
> 2. Every developer & CI machine will have pycommonbla in its site packages?
> 3. solvent bring pycommonbla to some directory, and add it to PYTHONPATH?
> (That's what I planned for strato_kv)
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Stratoscale/upseto/pull/2#issuecomment-115298816>.
>



-- 
Thanks,
Sasha.

File: upseto/packegg.py [hunk 1]
```diff
@@ -66,6 +73,9 @@ def addArgumentParserParameters(cls, parser):
             "--joinPythonNamespaces", action="store_true",
             help="collect files from joined namespaces. Only works when "
             "invoked from an upseto project directory")
+        parser.add_argument(
+            "--joinWithSitePackages", nargs="*", default=[],
+            help="Continue join those packages with site package")
```
Comment (lior-stratoscale) [inline upseto/packegg.py]: help message is unclear

**Pull request merged**